### PR TITLE
Generate a `.signature.manifest`

### PR DESCRIPTION
**Pull request by filiptronicek**
Generates a signature manifest with all the contents of a `vsix` along with each file's digest. At this time, no validation against the manifest is done, only generation.

It's JP's suggestion from https://github.com/filiptronicek/node-ovsx-sign/pull/16#issuecomment-2138302159.

## How to test

```
node lib/node-ovsx-sign key-pair
node lib/node-ovsx-sign sign some-extension.vsix private.pem
```

### CODE REVIEW

File: src/utils/signature-manifest.ts [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+
+/**
+ * @returns a sha256 hash of the file
+ */
+const fileSha256 = async (file: Buffer): Promise<string> => {
+    return new Promise((resolve, reject) => {
+        try {
+            const hash = createHash("sha256");
+            hash.update(file);
+            const digest = hash.digest("hex");
```
Comment (jeanp413) [inline src/utils/signature-manifest.ts:30]: digest is also base64 encoded

**Comment by jeanp413** [comment]
There's something tricky about the zip library I think or somewhere else, it's reporting filenames with lowercase always example `[content_types].xml'` and it should be `'[Content_Types].xml'`

File: src/utils/signature-manifest.test.ts [hunk 2]
```diff
@@ -0,0 +1,19 @@
+import { download } from "./download";
+import { unlink } from "fs/promises";
+import { generateSignatureManifest } from "./signature-manifest";
+
+jest.setTimeout(20_000);
+
+describe("signatureManifestTest", () => {
+    test("be able to generate a signature manifest file", async () => {
+        const extensionDestination = await download(
+            "https://open-vsx.org/api/jeanp413/open-remote-ssh/0.0.33/file/jeanp413.open-remote-ssh-0.0.33.vsix",
```
Comment (jeanp413) [inline src/utils/signature-manifest.test.ts:10]: Maybe better to use an extension from MS marketplace, so we are sure it matches the "oficial" manifest

**Comment by jeanp413** [comment]
Tested downloading python vsix and sigzip from MS marketplace

File: src/utils/signature-manifest.ts (hunk 1 above)
Comment (filiptronicek) [reply src/utils/signature-manifest.ts:30]: I wonder if it's just encoded directly into base64 or first hex and then base64. I'll try to reverse engineer this

Comment (filiptronicek) [reply src/utils/signature-manifest.ts]: It turns out it's the former.

File: src/utils/signature-manifest.test.ts (hunk 2 above)
Comment (filiptronicek) [reply src/utils/signature-manifest.test.ts:10]: Updated #18 to have tests against the official `vsix` and signature manifest. It turns out the unzipping function `vsce` includes transforms all file names into lowercase, which I have now patched.

**Pull request merged**